src/js/lib/content/src/index.js

```diff
@@ -179,20 +179,37 @@ const MAX_MESSAGE_LENGTH = 500;
 
 function remoteLog(event,data) {
     if (mixpanel) {
-        event = JSON.stringify(event).substr(0,MAX_MESSAGE_LENGTH)
-        if (data !== undefined) {
+        event = event.substr(0,MAX_MESSAGE_LENGTH)
+        if (data !== undefined && data !== null) {
             data = JSON.stringify(data).substr(0,MAX_MESSAGE_LENGTH)
+            mixpanel.track(event,{data});
+        } else {
+            mixpanel.track(event);
         }
-        mixpanel.track(event,{data:data});
     } else {
         console.warn(`Failed to log the event '${event}'.`)
     }
 }
 
-window.enso.remote_log = remoteLog
+function register(data) {
+    if (mixpanel) {
+        if (data !== undefined) {
+            data = JSON.stringify(data).substr(0,MAX_MESSAGE_LENGTH)
```

Comment:
I would add some indicator that the message was trimmed, like `+ ' ...'`.

Reply:
Added.